libmamba/include/mamba/core/url.hpp

```diff
@@ -41,6 +42,16 @@ extern "C"
 
 namespace mamba
 {
+    namespace conda_urls
+    {
+        // usernames on anaconda.org can have a underscore, which influences the
+        // first two characters
+        inline const std::regex token_regex{ "/t/([a-zA-Z0-9-_]{0,2}[a-zA-Z0-9-]*)" };
+        inline const std::regex http_basicauth_regex{ "(://|^)([^\\s]+):([^\\s]+)@" };
+        inline const std::regex scheme_regex{ "[a-z][a-z0-9]{0,11}://" };
+    }
```

Comment:
Should be move these definitions to the cpp file?

Reply:
Unfortunately that would mean potential static initialization fiasco, that's why I opted with `inline const` as it guarantees uniqueness in the binary (although it's possible 2 binaires have their own copy, it should always be the same value).

IFF we had `static constexpr` working for that type then it would be ideal (`constexpr` implies `inline` but it would also guarantee that these objects are built before runtime). Or `constinit` but that's C+20.
Another way to simplify all this would be to use [CTRE](https://compile-time-regular-expressions.readthedocs.io/en/latest/#) instead where we know the regex expression at compile-time, which would be ideal (in all aspects, including speed) but would mean an additional dependency.

Reply:
I'm all for CTRE! `std::regex` has bad rep

Reply:
@Klaim not sure I get it, these variables seem to be used in `url.cpp` functions only (although I might have missed some uses), and don't depend on each other, so I don't see how we can run in a static initialization fiasco.

Reply:
They were originally in `context.hpp` so I assumed that users want to have access to these?
If not then ok I'll move them in the cpp.

Reply:
See for details: https://github.com/mamba-org/mamba/pull/2615/files/536d56f2a48960678e5a7e21b392ed4bd29a072c#diff-b13b39da2b425585073993cc5e342e9b973c3863efbbc0fd760790d74cd89831L283-L290

Reply:
These regex are also used in `util.cpp` so I'll have to keep them that way.